torch/_higher_order_ops/utils.py

```diff
@@ -309,6 +309,22 @@ def fw_with_masks(*args):
     return fw_with_masks
 
 
+# This function replaces None gradients with all-zero gradients.
+# In PyTorch there is the convention that None gradients are
+# replaced with all-zeros gradients
+def unmask_none_gradients(grads, operands):
+    return [
+        g
+        if (
```

Comment:
Need some tighter constraints on the types of operand when g is None and what're the types that can be returned.

Reply:
I revised the constraints. What do you think?